ros_buildfarm/status_page.py

```diff
@@ -39,6 +39,111 @@
 from .templates import get_template_path
 
 
+def build_blocked_source_entries_page(
+        config_url, rosdistro_name,
+        output_dir, copy_resources=False):
+    from rosdistro import get_cached_distribution
+    from rosdistro import get_index
+    from rosdistro.dependency_walker import DependencyWalker
+
+    start_time = time.localtime()
+
+    config = get_config_index(config_url)
+    index = get_index(config.rosdistro_index_url)
+
+    rosdistro_name = _rosdistro_or_latest(index, rosdistro_name)
+    print('Getting blocked source entries for', rosdistro_name)
+
+    try:
+        prev_rosdistro_name = _prev_rosdistro(index, rosdistro_name)
+    except ValueError as e:
+        print(e.args[0], file=sys.stderr)
+        exit(-1)
+
+    print('Comparing', rosdistro_name, 'with', prev_rosdistro_name)
+
+    dist = get_cached_distribution(index, rosdistro_name)
+    prev_dist = get_cached_distribution(index, prev_rosdistro_name)
+
+    prev_walker = DependencyWalker(prev_dist)
+
+    prev_released_repos = set(_released_repos(prev_dist))
+    source_entry_repos = set(_source_entry_repos(dist))
+    missing_repos = prev_released_repos.difference(source_entry_repos)
+
+    # Assume repos will provide the same packages as previous distro
+    missing_packages = set(_released_packages(prev_dist, missing_repos))
+
+    repos_info = defaultdict(dict)
+
+    # Give all repos some basic info
+    for repo_name in prev_released_repos.union(source_entry_repos).union(missing_repos):
+        repos_info[repo_name]['url'] = ''
+        repos_info[repo_name]['repos_blocking'] = set()
+        repos_info[repo_name]['recursive_repos_blocking'] = set()
+        repos_info[repo_name]['released'] = False
+        repos_info[repo_name]['version'] = 'no'
+        repos_info[repo_name]['repos_blocked_by'] = {}
+        repos_info[repo_name]['maintainers'] = defaultdict(dict)
+
+    for repo_name in prev_released_repos:
+        repos_info[repo_name]['url'] = _repo_url(prev_dist, repo_name)
+
+    # has a source entry? Lets call that 'released' with a 'version' to reuse
+    # logic for the blocked releases page
+    for repo_name in source_entry_repos:
+        repos_info[repo_name]['released'] = True
+        repos_info[repo_name]['version'] = 'yes'
+
+    # Determine which repos directly block the missing ones
+    for repo_name in missing_repos:
+        package_dependencies = set()
+        for pkg in _released_packages(prev_dist, (repo_name,)):
+            package_dependencies.update(_package_dependencies(prev_walker, pkg))
+
+        for dep in package_dependencies:
+            if dep in missing_packages:
+                blocking_repo = prev_dist.release_packages[dep].repository_name
+                if blocking_repo == repo_name:
+                    # ignore packages in the same repo
+                    continue
+                url = _repo_url(prev_dist, repo_name)
+                repos_info[repo_name]['repos_blocked_by'][blocking_repo] = url
```

Comment:
I would suggest to avoid the single use variable `url` and directly assign the return value of the function call.

Same for `new_maintainers` below.

Reply:
Removed single use variables in 07d3052